exporter/collector/googlemanagedprometheus/extra_metrics_test.go

```diff
@@ -239,6 +242,52 @@ func TestAddExtraMetrics(t *testing.T) {
 				return metrics
 			}(),
 		},
+		{
+			name: "metric as double",
+			config: Config{ExtraMetricsConfig: ExtraMetricsConfig{
+				EnableScopeInfo:  true,
+				EnableTargetInfo: true,
+			}},
+			input:                   testMetric(timestamp),
+			enableDoubleFeatureGate: true,
+			expected: func() pmetric.ResourceMetricsSlice {
+				metrics := testMetric(timestamp).ResourceMetrics()
+				scopeMetrics := metrics.At(0).ScopeMetrics()
+
+				// Insert a new, empty ScopeMetricsSlice for this resource that will hold target_info
+				sm := scopeMetrics.AppendEmpty()
+				metric := sm.Metrics().AppendEmpty()
+				metric.SetName("target_info")
+				// This changes the value to double because of the feature gate.
+				metric.SetEmptyGauge().DataPoints().AppendEmpty().SetDoubleValue(1)
```

Comment:
should probably add a test case for Sum too, and one for the default/continue case